h5py/_hl/filters.py

```diff
@@ -259,7 +263,23 @@ def rq_tuple(tpl, name):
 
     if chunks is not None:
         plist.set_chunk(chunks)
-        plist.set_fill_time(h5d.FILL_TIME_ALLOC)  # prevent resize glitch
+
+    if fill_time is None:
+        if chunks is None:
+            fill_time = 'ifset'
+        else:
+            fill_time = 'alloc'  # prevent resize glitch
+    else:
+        invalid_fill_time = (isinstance(fill_time, str) and
+                             fill_time.lower() not in
```

Comment:
Our existing one-of-these-strings parameters, like `compression`, `driver` and `mode`, are case-sensitive, and I think this is a common pattern (e.g. the `byteorder=` parameter for `int.from_bytes`. So let's stick with that here.

Reply:
Adopted.